### ISSUE TITLE
Host marco didn't save on host edit

### ISSUE THREAD

**Issue by rk4an** [centreon/centreon#4951]
**Centreon Web version**:  2.8.3

**Centreon Engine version**:  1.7.0

**Centreon Broker version**:  

**OS**: Centreon 3.4.1

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Create a command with host macro
2. Edit host and set a value to host macro
3. Save

**Describe the results you received:**

Host not saved

**Describe the results you expected:**

Host save

**Additional information you think important (e.g. issue happens only occasionally):**

**Comment by anzevi** [comment]
try to manually restart the `centengine` service. I think it's the case of service not being reloaded/restarted. Having the same issue atm

**Comment by rk4an** [comment]
Hi, 

Thanks, same problem after restarting this service

**Comment by leoncx** [comment]
Hello,

Can you attach your sql-error.log ?

**Comment by rk4an** [comment]
Hi,

The file is empty, no error in SQL side

**Comment by thiuyendang** [comment]
Centreon 2.8.5 CentOS6 CentOS7
Editing a host
Add a new entry in Arguments section
Save
Edit again
Changes have been saved.

**Comment by larkoie** [comment]
I'm on centreon web 2.8.16 (CentOS7) and this issue is still here :
Edit host
Add new entry in CUSTOM MACROS (not argument as stated by thiuyendang)
Save
Edit again
Changes are lost

**Comment by lpinsivy** [comment]
Do you have error in Apache log file?
Do you have javascript error in the JS console of your browser?
How many entries do you have in centreon.on_demand_macro_host table?
What is the value of AUTO_INCREMENT of centreon.on_demand_macro_host table?

**Comment by larkoie** [comment]
Hi,
There are no error in Apache on the server and no JS error on the client browser.
No entries in the table "on_demand_macro_host"

I figured out the issue for me:
I was trying to add SNMPVERSION and SNMPCOMMUNITY macros on the host side while in fact I had to fill in the "SNMP Community & Version" fields in the "Host basic information" section. I guess that it sets those macros without showing them in the macro section and it prevent us to create new macros with the same name?
